### PR TITLE
Rest HL client: Add get license action

### PR DESCRIPTION
**Pull request by imotov**
WIP: @nik9000, @hub-cap I have implemented it in the license-as-a-string 
style that we discussed before, but just want to make sure that we are still
ok with it for this one, before polishing it off. Could you take a look when you 
have a chance?

Continues to use String instead of a more complex License class to
hold the license text similarly to put license.

Relates #29827

### CODE REVIEW

**Review by hub-cap — CHANGES_REQUESTED** [review]
I would like @nik9000 opinion here. I feel like not using a License class client side (and opting for a String instead) ended up making this code look less nice. And it puts a 1off (`convertResponseToJson`) in the client as well, which feels wrong. But id be happy to be told that im mistaken (spoiler alert, i usually am mistaken)!

File: x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/license/GetLicenseResponse.java
Comment (nik9000) [inline x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/license/GetLicenseResponse.java]: I'd be `getLicense`

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java
Comment (nik9000) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java]: I was happy with the put license API using the license json because I figured most people don't edit their license. It is just a big blob of json they get from somewhere and stick into wherever it needs to go. I mean, I'd have been happy with using the `License` object from x-pack:plugin:core but it has so much stuff in it! There is like a file watcher in there. All kinds of stuff that has no business being in a client at all. I feel like it'd be nice to get fields like max_nodes and expiry back in the client but License.java will take a lot of work to move. So I kind of figured it wasn't worth doing. I kind of wonder how much license administration folks will do with the java client, but :shrug:.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java
Comment (nik9000) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java]: Would it be ok for us to do `EntityUtils.toString` and leave it at that? Do we need to parse the response? If we make the request with the json content-type header we can be fairly sure it'll come back json. Right?

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java
Comment (imotov) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java]: How about this? I can move `convertResponseToJson` into LicenseClient, rename `public String license()` into `public String getLicenseDefinition()` and we will leave it like this for now. If we realize at some point that we actually want License as License, we can always implement a client-side version of the License class and add `public License getLicense()` method that will return it. This way we don't really paint ourselves into a corner and make progress.

Comment (nik9000) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java]: I'm fine with that. Though I still think you might be ok with `EntityUtils.toString` so you don't really need the helper at all.

**Comment by imotov** [comment]
@hub-cap, @nik9000 I have updated the PR as per our discussion. Could you take another look when you have a chance?

File: client/rest-high-level/src/main/java/org/elasticsearch/client/LicenseClient.java
Comment (nik9000) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/LicenseClient.java]: I think this belongs one line below.

**Review by nik9000 — APPROVED** [review]
LGTM.

**Review by hub-cap — APPROVED** [review]
:shipit:

**Comment by hub-cap** [comment]
I have just committed #32596 which changes the location of the xpack clients. Its very likely your tests wont compile now that the xpack portion of `client.xpack().yourCommercialClient()` is removed.

**Comment by imotov** [comment]
@hub-cap thanks for letting me know! I typically run precommit after merging in master for stale PRs, but since this PR was pretty fresh, I might have just pushed it since the change didn't cause any conflicts. 

I am waiting for #32613 to clear and will merge this PR in afterwards.

**Pull request merged**
